inc/Engine/CriticalPath/CriticalCSSSubscriber.php

```diff
@@ -488,6 +491,61 @@ public function warning_critical_css_dir_permissions() {
 		}
 	}
 
+	/**
+	 * Insert loadCSS script in <head>.
+	 *
+	 * @since  2.11.2 Updated loadCSS rel=preload polyfill to version 2.0.1
+	 * @since  2.10
+	 */
+	public function insert_load_css() {
+		if ( rocket_get_constant( 'DONOTROCKETOPTIMIZE' ) ) {
+			return;
+		}
+
+		if ( ! $this->options->get( 'async_css', 0 ) ) {
+			return;
+		}
+
+		if ( is_rocket_post_excluded_option( 'async_css' ) ) {
+			return;
+		}
+
+		if (
+			empty( $this->critical_css->get_current_page_critical_css() )
+			&&
+			empty( $this->options->get( 'critical_css', '' ) )
+		) {
+			return;
+		}
+
+		// This filter is documented in inc/front/process.php.
+		$rocket_cache_search = apply_filters( 'rocket_cache_search', false );
+
+		// Don't apply on search page.
+		if ( is_search() && ! $rocket_cache_search ) {
+			return;
+		}
+
+		// Don't apply on 404 page.
+		if ( is_404() ) {
+			return;
+		}
+
```

Comment:
```suggestion
//group posts related conditions
		if (
			(
				is_search()
				&&
				! apply_filters( 'rocket_cache_search', false )
			)
			||
			is_404()
			||
			is_rocket_post_excluded_option( 'async_css' )
		){
			return;
		}

		if (
			empty( $this->critical_css->get_current_page_critical_css() )
			&&
			empty( $this->options->get( 'critical_css', '' ) )
		){
			return;
		}
```
what about grouping conditions related to posts?

Reply:
`is_search()` is going to be the search results page, and `is_404()` can be anything, so they are not conditions tied to a post specifically.

Reply:
You could almost extract _all_ these checks into their own private method.

Reply:
@Tabrisrp Ok, we can reorder the conditions to put `get_current_page_critical_css` at the end as this is using the filesystem so it will be almost slower than checking database for example.
Also like @iCaspar stated, I can see that these conditions or most of them are used on `insert_critical_css_buffer` buffer.
be calm with our code review we are studying the code on that :D

Reply:
Good idea for the private method, done here: https://github.com/wp-media/wp-rocket/pull/2895/commits/5415787628da77aa59ee73a3c69e8a8635c2e15e